apps/studio/src/server/modules/permissions/permissions.type.ts

```diff
@@ -0,0 +1,20 @@
+import type { PureAbility } from "@casl/ability"
+
+import type { Resource as RawResource } from "../database"
+
+type Resource = Pick<RawResource, "parentId">
+
+type AllowedResourceActions = (typeof ALL_ACTIONS)[number]
+export type CrudResourceActions = (typeof CRUD_ACTIONS)[number]
```

Comment:
nit: can define these 2 after the respective declarations

Reply:
sgtm!